frb_codegen/src/config.rs

```diff
@@ -1,70 +1,181 @@
-#[structopt(setting(AppSettings::DeriveDisplayOrder))]
+#[derive(Parser, Debug, PartialEq, Deserialize, Default)]
+#[clap(version, setting(clap::AppSettings::DeriveDisplayOrder))]
 pub struct RawOpts {
     /// Path of input Rust code
-    #[structopt(short, long)]
+    #[clap(short, long, required = true, multiple_values = true)]
     pub rust_input: Vec<String>,
     /// Path of output generated Dart code
-    #[structopt(short, long)]
+    #[clap(short, long, required = true, multiple_values = true)]
     pub dart_output: Vec<String>,
     /// If provided, generated Dart declaration code to this separate file
-    #[structopt(long)]
+    #[clap(long)]
     pub dart_decl_output: Option<String>,
-
-    /// Path of output generated C header
-    #[structopt(short, long)]
+    /// Output path of generated C header
+    #[clap(short, long)]
     pub c_output: Option<Vec<String>>,
     /// Crate directory for your Rust project
-    #[structopt(long)]
+    #[clap(long, multiple_values = true)]
     pub rust_crate_dir: Option<Vec<String>>,
-    /// Path of output generated Rust code
-    #[structopt(long)]
+    /// Output path of generated Rust code
+    #[clap(long, multiple_values = true)]
     pub rust_output: Option<Vec<String>>,
     /// Generated class name
-    #[structopt(long)]
+    #[clap(long, multiple_values = true)]
     pub class_name: Option<Vec<String>>,
-    /// Line length for dart formatting
-    #[structopt(long)]
-    pub dart_format_line_length: Option<i32>,
+    /// Line length for Dart formatting
+    #[clap(long, default_value = "80")]
+    pub dart_format_line_length: u32,
     /// Skip automatically adding `mod bridge_generated;` to `lib.rs`
-    #[structopt(long)]
+    #[clap(long)]
     pub skip_add_mod_to_lib: bool,
     /// Path to the installed LLVM
-    #[structopt(long)]
+    #[clap(long, multiple_values = true)]
     pub llvm_path: Option<Vec<String>>,
     /// LLVM compiler opts
-    #[structopt(long)]
+    #[clap(long)]
     pub llvm_compiler_opts: Option<String>,
     /// Path to root of Dart project, otherwise inferred from --dart-output
-    #[structopt(long)]
+    #[clap(long, multiple_values = true)]
     pub dart_root: Option<Vec<String>>,
-    /// Skip running build_runner even when codegen-capable code is detected
-    #[structopt(long)]
+    /// Skip running build_runner even when codegen-required code is detected
+    #[clap(long)]
     pub no_build_runner: bool,
     /// Show debug messages.
-    #[structopt(short, long)]
+    #[clap(short, long)]
     pub verbose: bool,
+    /// Enable WASM module generation.
+    /// Requires: --dart-decl-output
+    #[clap(long, requires = "dart-decl-output")]
+    pub wasm: bool,
+}
+
+/// Generic accumulator over the targets.
+///
+/// [`Acc<Option<String>>`] implements [`From<T>`] where <code>T: [ToString]</code>
+/// for code shared between all platforms.
+#[derive(Debug, Default, Clone)]
+pub struct Acc<T> {
```

Comment:
I wonder whether it used in `config.rs` or solely? If not, or not solely for configurations, maybe we can extract it to utils.rs etc

Btw nice abstraction :)